packages/react/src/auth/useSendSignInLinkToEmailMutation.ts

```diff
@@ -0,0 +1,24 @@
+import { useMutation, type UseMutationOptions } from "@tanstack/react-query";
+import {
+  type ActionCodeSettings,
+  type Auth,
+  AuthError,
+  sendSignInLinkToEmail,
+} from "firebase/auth";
+
+type AuthUseMutationOptions<TData = unknown, TError = Error> = Omit<
+  UseMutationOptions<TData, TError, void>,
+  "mutationFn"
+>;
+
+export function useSendSignInLinkToEmailMutation(
+  auth: Auth,
+  email: string,
+  actionCodeSettings: ActionCodeSettings,
+  options?: AuthUseMutationOptions
+) {
+  return useMutation<void, AuthError, void>({
+    ...options,
+    mutationFn: () => sendSignInLinkToEmail(auth, email, actionCodeSettings),
```

Comment:
This approach needs updating for mutations, the actual values need to come from the mutation call, not the hook. At the moment the user would need to do the following:

```js
const send = useSendSignInLinkToEmailMutation(auth, 'email', { ... });

...

function onSubmit() {
  send.mutate();
}
```

When actually, those values need to be reactive and passed to the mutation when called:

```js
const send = useSendSignInLinkToEmailMutation(auth);

...

function onSubmit(e) {
  const data = new FormData(e.currentTarget);
  send.mutate({ email: data.get('email'), actionCodeSettings: { ... });
}
```